main/src/ca/uwaterloo/flix/language/phase/unification/BoolUnification.scala

```diff
@@ -29,6 +29,15 @@ object BoolUnification {
    * Returns the most general unifier of the two given Boolean formulas `tpe1` and `tpe2`.
    */
   def unify(tpe1: Type, tpe2: Type, renv0: RigidityEnv)(implicit flix: Flix): Result[(Substitution, List[Ast.BroadEqualityConstraint]), UnificationError] = {
+
+    // Set the variable levels to the minimum of all flexible variables involved.
```

Comment:
I think this code is duplicated. Maybe we can move it into a helper function in Levels? Since it concerns that mostly?